test/integration/Integration.t.sol

```diff
@@ -0,0 +1,640 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.23;
+
+import { Test } from "../../lib/forge-std/src/Test.sol";
+
+import { Proxy } from "../../src/Proxy.sol";
+
+import { WrappedMToken } from "../../src/WrappedMToken.sol";
+
+interface IRegistrarLike {
+    function addToList(bytes32 list, address account) external;
+
+    function removeFromList(bytes32 list, address account) external;
+
+    function setKey(bytes32 key, bytes32 value) external;
+}
+
+interface IMTokenLike {
+    function approve(address spender, uint256 amount) external returns (bool success);
+
+    function transfer(address recipient, uint256 amount) external returns (bool success);
+
+    function balanceOf(address account) external view returns (uint256 balance);
+
+    function currentIndex() external view returns (uint128 index);
+
+    function isEarning(address account) external view returns (bool earning);
+
+    function totalEarningSupply() external view returns (uint240 supply);
+
+    function totalNonEarningSupply() external view returns (uint240 supply);
+}
+
+contract IntegrationTests is Test {
+    IMTokenLike internal constant _mToken = IMTokenLike(0x866A2BF4E572CbcF37D5071A7a58503Bfb36be1b);
+
+    address internal constant _minterGateway = 0xf7f9638cb444D65e5A40bF5ff98ebE4ff319F04E;
```

Comment:
Unused.
```suggestion
```

Reply:
Wil be used when new integration tests are added that actually mint more M.